src/pages/en/guides/styling.md

```diff
@@ -92,6 +92,42 @@ const backgroundColor = "rgb(24 121 78)";
 
 📚 See our [directives reference](/en/reference/directives-reference/#definevars) page to learn more about `define:vars`.
 
+### Passing a `class` to a child component
+
+In Astro, HTML attributes like `class` do not automatically pass through to child components.
+
+
+Instead, accept a `class` prop in the child component and apply it to the root element:
+
+```astro title="src/components/MyComponent.astro" {1}
+<div class={Astro.props.class}>
```

Comment:
Happier now with these examples! Just pointing out that I don't think we show `Astro.props` anywhere else in Docs as an attribute directly. We *always* have a `const { class } = Astro..props` destructuring statement in frontmatter, as far as I can remember.

I'm not opposed to this, and I know it makes the example more concise. But I worry about having the only time we show this be in one of these "you can't do this, you have to do this!" kind of situations. If we had introduced in a more general passing props intro-y example, then this wouldn't stand out as, "Oh, maybe for exactly this situation, I need to do this?"

Thoughts?

Reply:
It _is_ something you need for just this situation! It's one of two workarounds for `class` being a reserved word: you either destructure and rename (second example) or leave it on the `props` object (this first example).

Reply:
If it's confusing, I could just remove it and keep the second workaround.

Reply:
Ah, I totally see that now, that they are two versions of the same thing! I kind of... want to just keep the second example? It seems more in keeping with the rest of the Docs? We still never show applying something from props directly like that.

OR, I'd show the second one first, since that is very Astro-Docs-like. Then, show this as an example of avoiding needing to rename, and THIS becomes the, "Hey, you can actually do this, too, skipping the frontmatter" example that doesn't otherwise show up in Docs.